docs/Stream_Transformation_Functions.md

```diff
@@ -0,0 +1,52 @@
+# Stream transformation functions
+
+Create new stream transformation functions in Go. 
+
+## Required guidelines
+
+- Stream transformation functions must conform to the examples in [universe](https://github.com/influxdata/flux/blob/master/stdlib/universe) and include the required function and methods shown in the table below.
+- You must submit a unit test in in the same folder as the new stream transformation function implementation.
+- You must submit an end-to-end test in [testdata](https://github.com/influxdata/flux/tree/master/stdlib/testing/testdata). Please look at [End_to_End_Testing.md](https://github.com/influxdata/flux/tree/master/docs/End_to_End_Testing.md) for details.
+- You must add a description of the function to [SPEC.md](./docs/SPEC.md).
+
+### Required function and methods
+
+1. **init():**
+
+2. 1. Where you define your function signature and register the methods(**Op-Spec**, **Procedure Spec**, **Transformation Spec**)
+
+3. **FunctionSignature**: 
+
+4. 1. Where you define how you want the user to write out a function in flux and what inputs they need to include. 
+
+   2. For example, with timeShift, this is where you would define that a user must provide a duration and a time columns. 
+
+   3. 1. \```|>timeShift(duration: 10h, columns: ["_start", "_stop", "_time"])```
+
+5. **Op-Spec:**
+
+6. 1. The internal representation or mirror of the function signature. 
+   2. The **Op-Spec** defines the function signature and gets converted converted into the **Procedure Spec**. 
+   3. The **Op-Spec** identifies and collects function arguments and then encodes them into a JSON-encodable struct. 
+
+7. **Procedure Spec:**
+
+8. 1. Identifies the incoming **Op-Spec**. 
+   2. Copies incoming values and converts them if needed.
+   3. Creates the plan for the actual transformation of the data. It’s optimized by the query planner which passes the plan into the execution engine after and converted into the Transformation type. 
+
+9. **Transformation Spec:**
```

Comment:
-`Transformation Spec`(hyphen?): Bulk of the script responsible for transforming data.